src/coap_openssl.c

```diff
@@ -2806,11 +2806,15 @@ void * coap_dtls_new_server_session(coap_session_t *session) {
   /* hint may get updated if/when handling SNI callback */
   psk_hint = coap_get_session_server_psk_hint(session);
   if (psk_hint != NULL && psk_hint->length) {
-    char hint[psk_hint->length+1];
+    char* hint = OPENSSL_malloc(psk_hint->length + 1);
 
-    memcpy (hint, psk_hint->s, psk_hint->length);
-    hint[psk_hint->length] = '\000';
-    SSL_use_psk_identity_hint(ssl, hint);
+    if (hint) {
+      memcpy(hint, psk_hint->s, psk_hint->length);
+      hint[psk_hint->length] = '\000';
+      SSL_use_psk_identity_hint(ssl, hint);
+      OPENSSL_free(hint);
+    } else
+      coap_log(LOG_WARNING, "hint malloc failure\n");
```

Comment:
`coap_log()` is a pre-processor macro. Therefore, and in general, I suggest including it in a block.

Reply:
Updated, and code pushed.